src/data/set/n_ary.lean

```diff
@@ -0,0 +1,289 @@
+/-
+Copyright (c) 2020 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import data.set.image
+
+/-!
+# N-ary images of sets
```

Comment:
Can you split this to a standalone PR? It looks like it's more isolated than the rest of the changes.